wallet/src/bin/random_wallet.rs

```diff
@@ -196,8 +205,9 @@ async fn main() {
             asset.definition
         }
         None => {
+            let policy = AssetPolicy::default().set_freezer_pub_key(freeze_key.clone());
```

Comment:
You must also audit in order to be able to freeze. Otherwise the wallet won't be able to discover any records that it can freeze. Also note that the policy has to reveal everything: amount, address, and blind.